configure.ac

```diff
@@ -3247,6 +3247,93 @@ case "$host" in
 		;;
 esac
 
+PAL_SUPPORTED=no
+PAL_ARCH=
+PAL_OS=
+
+case "$host" in
+  i*86-*-*)
+	case $host_os in
+	  linux*)
+		PAL_SUPPORTED=yes
+		PAL_ARCH=x86
+		PAL_OS=linux
+		;;
+	  darwin*)
+		PAL_SUPPORTED=yes
+		PAL_ARCH=x86
+		PAL_OS=osx
+		;;
+	  freebsd*|kfreebsd-gnu*)
+		PAL_SUPPORTED=yes
+		PAL_ARCH=x86
+		PAL_OS=freebsd
+		;;
+	esac
+	;;
+  x86_64-*-* | amd64-*-*)
+	case $host_os in
+	  linux*)
+		PAL_SUPPORTED=yes
+		PAL_ARCH=x64
+		PAL_OS=linux
+		;;
+	  darwin*)
+		PAL_SUPPORTED=yes
+		PAL_ARCH=x64
+		PAL_OS=osx
+		;;
+	  freebsd*|kfreebsd-gnu*)
+		PAL_SUPPORTED=yes
+		PAL_ARCH=x64
+		PAL_OS=freebsd
+		;;
+	esac
+	;;
+  armv*-*-*)
+	case $host_os in
+	  darwin*)
+		PAL_SUPPORTED=yes
+		PAL_ARCH=arm
+		PAL_OS=osx
```

Comment:
Same as for android, the System.native build script doesn't have a parameter for anything else than `freebsd`, `linux`, `netbsd` and `osx`.